### PR TITLE
[ENGG-2116] feat: import and export for mocks 

### PR DESCRIPTION
**Pull request by rohanmathur91**
Changelog:
- Added export support for mocks
- Bulk option for export 
- Added import mocks support

Demo:

 https://github.com/user-attachments/assets/9745a47c-7193-4d7a-b2e3-b6afc2c34e53
 
 **PR Summary by Typo**
------------

 **Summary**

This pull request introduces new features for exporting and importing mocks in the MocksList component. It includes the creation of new components and functions, as well as updates to existing ones.

**Key Points**

1. New actions `exportMocksAction` and `importMocksAction` are added to the `MocksActionContextProvider`.
2. A new component `DownloadMocksModalWrapper` is introduced to manage the state for opening and handling the `DownloadMocksModal`.
3. A new utility function `prepareMocksToExport` is created to clean and prepare mocks for export.
4. The ImportMocksModal component is implemented to allow users to import mocks through a file picker.
5. Analytics events are tracked for exporting and importing mocks. 

 <h6>To turn off PR summary, please visit <a href="https://app.typoapp.io/settings/dev-analytics/notification?tab=codeHealth">Notification settings</a>.</h6>

### CODE REVIEW

**Comment by rohanmathur91** [comment]
Closes: https://github.com/requestly/requestly/issues/2128

**Review by nsrCodes — CHANGES_REQUESTED** [review]

File: app/src/features/mocks/modals/ImportMocksModalWrapper/utils.ts
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/utils.ts:48]: Added a logger, toast part is handled outside in component

File: app/src/features/mocks/modals/ImportMocksModalWrapper/utils.ts
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/utils.ts:20]: This is handled, it checks for `result.success` before proceeding further, else throws an error.

File: app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/hooks/useMocksTableColumns.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/hooks/useMocksTableColumns.tsx:235]: The record type we get in table is `RQMockSchema` and when dealing with collection we need some other properties, checking `isCollection` wont and then proceeding wont work.

File: app/src/features/mocks/modals/DownloadMocksModalWrapper/DownloadMocksModal.scss
Comment (rohanmathur91) [reply app/src/features/mocks/modals/DownloadMocksModalWrapper/DownloadMocksModal.scss]: For now not changing this, it will be done for all the screens at once

File: app/src/features/mocks/modals/DownloadMocksModalWrapper/DownloadMocksModal.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/DownloadMocksModalWrapper/DownloadMocksModal.tsx]: This wont happen, just a precaution

File: app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx:120]: It might get to nested

File: app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx:239]: This is what we are using for other import modals too, can be pick separately

File: app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/utils/index.ts
Comment (rohanmathur91) [reply app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/utils/index.ts]: This are used in js files too where record might not be there, so just to prevent that

File: app/src/features/mocks/modals/ExportMocksModalWrapper/ExportMocksModal.tsx
Comment (nsrCodes) [inline app/src/features/mocks/modals/ExportMocksModalWrapper/ExportMocksModal.tsx:52]: I am sorry, I was not aware `null` is a valid type for this state when I wrote [this comment](https://github.com/requestly/requestly/pull/2123#discussion_r1740723906)

can you please bring back the old change

File: app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/utils/index.ts
Comment (nsrCodes) [reply app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/utils/index.ts]: I don't see this currently being used in any `js` file

File: app/src/features/mocks/modals/ImportMocksModalWrapper/utils.ts
Comment (nsrCodes) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/utils.ts:49]: or, as I now mention in a comment in `ImportMocksModal.tsx` and in [the comment before](https://github.com/requestly/requestly/pull/2123#discussion_r1740576846), just throw the error and let the consumer handle it.

File: app/src/features/mocks/modals/DownloadMocksModalWrapper/DownloadMocksModal.tsx
Comment (nsrCodes) [reply app/src/features/mocks/modals/DownloadMocksModalWrapper/DownloadMocksModal.tsx]: Redundant precautions make the code harder to read later

File: app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx
Comment (nsrCodes) [inline app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx:28]: [previous comment](https://github.com/requestly/requestly/pull/2123#discussion_r1740605617) has not been addressed here

File: app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx
Comment (nsrCodes) [inline app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx:24]: I am hoping the [previous comment](https://github.com/requestly/requestly/pull/2123#discussion_r1740620075) was accidentally marked as resolved, since this still does not seem required.

This is building on top of the state issue I mentioned above, so IMO it would be better to not add more (_especially unnecessary_) code on top of that.

**Review by nsrCodes — CHANGES_REQUESTED** [review]
please create relevant tickets and mark them as related to the current one. 
here's links to the relevant discussion comments 

[1](https://github.com/requestly/requestly/pull/2123#discussion_r1740582634) [2](https://github.com/requestly/requestly/pull/2123#discussion_r1741709361) [3](https://github.com/requestly/requestly/pull/2123#discussion_r1741713212) 
[4](https://github.com/requestly/requestly/pull/2123#discussion_r1740709244)  IMO, this and similar changes I mention here are small fixes and can be done before this PR gets merged.

File: app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx
Comment (nsrCodes) [reply app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx:7]: could also create a ticket for this as well (its a tech debt we have been avoiding)

File: app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx:120]: I feel `async await` is more readable then `.then`

File: app/src/features/mocks/modals/DeleteRecordsModalWrapper/DeleteRecordsModal/index.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/DeleteRecordsModalWrapper/DeleteRecordsModal/index.tsx:35]: [Refer](https://github.com/requestly/requestly/pull/2123#discussion_r1743271889)

File: app/src/features/mocks/modals/DownloadMocksModalWrapper/DownloadMocksModal.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/DownloadMocksModalWrapper/DownloadMocksModal.tsx]: Not needed, keeping it as it is

File: app/src/features/mocks/modals/ExportMocksModalWrapper/ExportMocksModal.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ExportMocksModalWrapper/ExportMocksModal.tsx:52]: Not needed, keeping it as it is

File: app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/hooks/useMocksTableColumns.tsx
Comment (nsrCodes) [reply app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/hooks/useMocksTableColumns.tsx:216]: okay, let this comment stay unresolved then. this isn't right (regardless of if this PR get's merged)
and at least an internal ticket should be logged to fixed this

Same for all the other issues that you marked as resolved even though they weren't addressed. 
- The ideal practice is to let the reviewer close the issue when they feel the issue is fixed. 
- Even if the PR is merged it serves as reference for later regarding what are known limitations of a change.

File: app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/hooks/useMocksTableColumns.tsx
Comment (nsrCodes) [reply app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/hooks/useMocksTableColumns.tsx:234]: refer my reply to this - https://github.com/requestly/requestly/pull/2123/files#r1743338023

File: app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/utils/index.ts
Comment (nsrCodes) [inline app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/utils/index.ts:10]: [this comment](https://github.com/requestly/requestly/pull/2123#discussion_r1742758683) seems to have been removed

Even though you have given an explanation of the cause [here](https://github.com/requestly/requestly/pull/2123#discussion_r1743271889), it does not justify NOT FIXING this right now

**Review by nsrCodes — CHANGES_REQUESTED** [review]
- let the reviewer mark his comments as resolved
- I am just trying to aim at the fact that underlying types should be resolved first before we start giving offline export based on code built on top of broken types
- any severe tech debt that we noticed should be logged internally to make sure that it gets fixed eventually

**Review by nsrCodes — APPROVED** [review]
I have added a [final thorough](https://github.com/requestly/requestly/pull/2123#pullrequestreview-2281852123) review as well but this has been blocked for too long so approving now

File: app/src/features/mocks/modals/ImportMocksModalWrapper/index.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/index.tsx:26]: https://linear.app/requestly/issue/ENGG-2265/cleanup-forcerender-in-mocks

File: app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/hooks/useMocksTableColumns.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksTable/hooks/useMocksTableColumns.tsx:234]: https://linear.app/requestly/issue/ENGG-2266/fix-mockmetadataschema-types

File: app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx]: https://linear.app/requestly/issue/ENGG-2267/bubble-up-mock-firebase-function-errors

File: app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ImportMocksModalWrapper/ImportMocksModal.tsx:226]: https://linear.app/requestly/issue/ENGG-2268/remove-redundant-classes-from-import-modals

File: app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx
Comment (rohanmathur91) [reply app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx:7]: https://linear.app/requestly/issue/ENGG-2269/refactor-modal-setup-for-rules-and-mocks

**Pull request merged**
